open-craft/frontend-app-gradebook
# fix: add eshe_instructor to allowedRoles

## Description

eSHE Instructors can't view the gradebook without this.

Port of #1 to Redwood.

## Supporting information

#1

## Testing instructions

1. Use open-craft/edx-platform#666 branch (`0x29a/bb8914/redwood-code-drift`) of edx-platform.
2. Enable feature `ENABLE_ESHE_INSTRUCTOR_ROLE` (see below for example Tutor plugin).
3. Enroll as a student in a course.
4. Enroll a non-staff user as an eSHE instructor:
   1. Go to course.
   2. Go to Instructor tab.
   3. Go to section "Course Team Management".
   4. Select role "eSHE Instructor", input username and enroll.
5. Log in as the user enrolled in step 4.
6. Go to course.
7. Go to Instructor tab.
8. Go to "Student Admin" tab.
9. Click "View Gradebook".
10. See that you can view the enrolled student from step 3 in the gradebook.

<details>
<summary>Example Tutor plugin to enable required feature</summary>

```
name: eshe
version: 0.1.0
patches:
  openedx-common-settings: |
    FEATURES["ENABLE_ESHE_INSTRUCTOR_ROLE"] = True
```

</details>


## Deadline

2024-07-22

## Other information

Private-ref: https://tasks.opencraft.com/browse/BB-8934
